### PR TITLE
Add support for grpc health check

### PR DESCRIPTION
**Pull request by averzicco**
Hi, This PR is to add support for [gRPC health check](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) to blackbox exporter.
I've read the discussion on issues #665 and #603, it seems that gRPC is out of scope for blackbox_exporter, if this is still the case just close the PR.

### CODE REVIEW

**Comment by averzicco** [comment]
great, I've checked the dev summit video and realized didn't implement the logic to specify the service name for the health check. I've just pushed the code which takes care of that

File: prober/grpc.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// Health defines a health-check connection.
+type Health interface {
```
Comment (roidelapluie) [inline prober/grpc.go:36]:
```suggestion
type GRPCHealthCheck interface {
```

I'd like to have GRPC in the name.

**Review by roidelapluie — APPROVED** [review]
LGTM besides name change.

File: prober/grpc.go (hunk 1 above)
Comment (averzicco) [reply prober/grpc.go]: makes sense, changed

**Comment by roidelapluie** [comment]
Thanks!

**Comment by autumnw** [comment]
Appreciate to all the people who joined conversation and contribute code here! This is so awesome! 
Kudos to @averzicco!

**Pull request merged**